src/main/java/cofh/thermaldynamics/duct/fluid/DuctUnitFluid.java

```diff
@@ -383,7 +383,7 @@ public int fill(FluidStack resource, boolean doFill) {
 			@Override
 			public FluidStack drain(FluidStack resource, boolean doDrain) {
 
-				if (isOpen(from)) {
+				if (isOutput(from.ordinal())) {
```

Comment:
context: Servos use `ConnectionType.BLOCKED` and `isOpen()` will always return false if the connectiontype does not allow transfer. We need to check for isOutput() instead to control draining the grid from this side.